datalad_container/containers_run.py

```diff
@@ -133,39 +144,51 @@ def __call__(cmd, container_name=None, dataset=None,
             # just prepend and pray
             cmd = container['path'] + ' ' + cmd
 
+        assume_ready = assume_ready or []
         extra_inputs = []
-        for extra_input in ensure_iter(container.get("extra-input",[]), set):
-            try:
-                xi_kwargs = dict(
-                    img_dspath=image_dspath,
-                    img_dirpath=op.dirname(image_path) or ".",
-                )
-                extra_inputs.append(extra_input.format(**xi_kwargs))
-            except KeyError as exc:
-                yield get_status_dict(
-                    'run',
-                    ds=ds,
-                    status='error',
-                    message=(
-                        'Unrecognized extra_input placeholder: %s. '
-                        'See containers-add for information on known ones: %s',
-                        exc,
-                        ", ".join(xi_kwargs)))
-                return
+        if not "extra-inputs" in assume_ready:
+            for extra_input in ensure_iter(container.get("extra-input",[]), set):
+                try:
+                    xi_kwargs = dict(
+                        img_dspath=image_dspath,
+                        img_dirpath=op.dirname(image_path) or ".",
+                    )
+                    extra_inputs.append(extra_input.format(**xi_kwargs))
+                except KeyError as exc:
+                    yield get_status_dict(
+                        'run',
+                        ds=ds,
+                        status='error',
+                        message=(
+                            'Unrecognized extra_input placeholder: %s. '
+                            'See containers-add for information on known ones: %s',
+                            exc,
+                            ", ".join(xi_kwargs)))
+                    return
+        assume_ready = [ar for ar in assume_ready if ar != 'extra-inputs']
+
+        if not "image" in assume_ready:
+            extra_inputs.append(image_path)
+        assume_ready = [ar for ar in assume_ready if ar != 'image']
 
         lgr.debug("extra_inputs = %r", extra_inputs)
 
+        # we can assume "both" with params choices and filtering above
+        if len(assume_ready) > 1:
+            assume_ready='both'
```

Comment:
Thanks for taking a stab at it! Just having a quick initial peek at it.

we will need some test for it, may be based on already existing one(s).

Note here that it should be a list I think, 
```suggestion
            assume_ready = ['both']
```
but I don't like such an implicit check/mapping... 

```shell
❯ python3 -c 'import this' | grep -e Zen -e Ex
The Zen of Python, by Tim Peters
Explicit is better than implicit.
```

Reply:
Agreed, made it more explicit, also I think `datalad.core.local.run._prep_worktree` takes a string ( inputs/outputs/both ) rather than a list, so I added a bit of logic to ensure that value.